File: URI/JavaScript/uri-1049.js
// URI - 1049: https://www.urionlinejudge.com.br/judge/pt/problems/view/1049

const input = require('fs').readFileSync('/dev/stdin', 'utf-8');
const characteristics = input.split('\n');

const eagleCharacteristics = ['vertebrado', 'ave', 'carnivoro'];
const doveCharacteristics = ['vertebrado', 'ave', 'onivoro'];
const humanCharacteristics = ['vertebrado', 'mamifero', 'onivoro'];
const cowCharacteristics = ['vertebrado', 'mamifero', 'herbivoro'];
const fleaCharacteristics = ['invertebrado', 'inseto', 'hematofago'];
const wormCharacteristics = ['invertebrado', 'inseto', 'herbivoro'];
const leechCharacteristics = ['invertebrado', 'anelideo', 'hematofago'];
const earthWormCharacteristics = ['invertebrado', 'anelideo', 'onivoro'];

console.log(searchForAnimalWithTheseCharacteristics(characteristics));

function searchForAnimalWithTheseCharacteristics(characteristics) {
  let animal = checkIfIsVertebrateAnimal(characteristics);
  if (animal)
    return animal;
  else if (animal = checkIfIsInvertebrateAnimal(characteristics))
    return animal;
  else
    return 'Not found.'
}

function checkIfIsVertebrateAnimal(characteristics) {
  if (checkCharacteristics(characteristics, eagleCharacteristics))
    return 'aguia';
  else if (checkCharacteristics(characteristics, doveCharacteristics))
    return 'pomba';
  else if (checkCharacteristics(characteristics, humanCharacteristics))
    return 'homem';
  else if (checkCharacteristics(characteristics, cowCharacteristics))
    return 'vaca';
}

function checkIfIsInvertebrateAnimal(characteristics) {
  if (checkCharacteristics(characteristics, fleaCharacteristics))
    return 'pulga';
  else if (checkCharacteristics(characteristics, wormCharacteristics))
    return 'lagarta';
  else if (checkCharacteristics(characteristics, leechCharacteristics))
    return 'sanguessuga';
  else if (checkCharacteristics(characteristics, earthWormCharacteristics))
    return 'minhoca';
}

function checkCharacteristics(receivedCharacteristics, animalCharacteristics) {
  return animalCharacteristics.includes(receivedCharacteristics[0]) &&
         animalCharacteristics.includes(receivedCharacteristics[1]) &&
         animalCharacteristics.includes(receivedCharacteristics[2]);
}

File: URI/Ruby/uri-1049.rb
# frozen_string_literal: true

# URI - 1049: https://www.urionlinejudge.com.br/judge/pt/problems/view/1049

def search_for_animal(characteristics)
  if characteristics[0] == 'vertebrado'
    search_for_vertebrate(characteristics)
  elsif characteristics[0] == 'invertebrado'
    search_for_invertebrate(characteristics)
  end
end

def search_for_vertebrate(characteristics)
  if characteristics[1] == 'ave'
    search_for_bird(characteristics)
  elsif characteristics[1] == 'mamifero'
    search_for_mammal(characteristics)
  end
end

def search_for_bird(characteristics)
  if characteristics[2] == 'carnivoro'
    'aguia'
  elsif characteristics[2] == 'onivoro'
    'pomba'
  end
end

def search_for_mammal(characteristics)
  if characteristics[2] == 'onivoro'
    'homem'
  elsif characteristics[2] == 'herbivoro'
    'vaca'
  end
end

def search_for_invertebrate(characteristics)
  if characteristics[1] == 'inseto'
    search_for_insect(characteristics)
  elsif characteristics[1] == 'anelideo'
    search_for_annelid(characteristics)
  end
end

def search_for_insect(characteristics)
  if characteristics[2] == 'hematofago'
    'pulga'
  elsif characteristics[2] == 'herbivoro'
    'lagarta'
  end
end

def search_for_annelid(characteristics)
  if characteristics[2] == 'hematofago'
    'sanguessuga'
  elsif characteristics[2] == 'onivoro'
    'minhoca'
  end
end

first_characteristic = gets.chomp
second_characteristic = gets.chomp
third_characteristic = gets.chomp
puts search_for_animal [first_characteristic, second_characteristic, third_characteristic]
